analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -197,89 +197,95 @@ private boolean canCombineWithLastRegisteredActivity(ActivityPayload payload, Ac
 		return true;
 	}
 
-	private boolean precedesLastRegisteredActivity(ActivityPayload payload, Activity lastRegisteredActivity)
+	private boolean precedesLastRegisteredActivity(ActivityPayload payload, ActivityDTO lastRegisteredActivity)
 	{
 		return payload.startTime.isBefore(lastRegisteredActivity.getStartTime());
 	}
 
-	private boolean isBeyondCombineIntervalWithLastRegisteredActivity(ActivityPayload payload, Activity lastRegisteredActivity)
+	private boolean isBeyondCombineIntervalWithLastRegisteredActivity(ActivityPayload payload, ActivityDTO lastRegisteredActivity)
 	{
 		ZonedDateTime intervalEndTime = lastRegisteredActivity.getEndTime()
 				.plus(yonaProperties.getAnalysisService().getConflictInterval());
 		return payload.startTime.isAfter(intervalEndTime);
 	}
 
-	private DayActivityCacheResult getRegisteredDayActivity(ActivityPayload payload, UserAnonymizedDTO userAnonymized,
-			GoalDTO matchingGoal)
+	private ActivityCacheResult getRegisteredDayActivity(ActivityPayload payload, GoalDTO matchingGoal)
 	{
-		DayActivity lastRegisteredDayActivity = cacheService.fetchLastDayActivityForUser(userAnonymized.getID(),
+		ActivityDTO lastRegisteredActivity = cacheService.fetchLastActivityForUser(payload.userAnonymized.getID(),
 				matchingGoal.getID());
-		if (lastRegisteredDayActivity == null)
+		if (lastRegisteredActivity == null)
 		{
-			return DayActivityCacheResult.none();
+			return ActivityCacheResult.cacheEmpty();
 		}
-		else if (isOnPrecedingDay(payload, lastRegisteredDayActivity))
+		else if (hasLaterDay(lastRegisteredActivity, payload))
 		{
-			// Fetch from the database because it is no longer in the cache
-			return DayActivityCacheResult.preceding(dayActivityRepository.findOne(userAnonymized.getID(),
-					getStartOfDay(payload.startTime, userAnonymized).toLocalDate(), matchingGoal.getID()));
+			return ActivityCacheResult.cachedHasLaterDay();
```

Comment:
Speaking about the names: can we rename this into ``createInstanceForTooNew``, and the other one into ``createInstanceForTooOld``?

Please rename the enum values of ``ActivityCacheResultStatus`` accordingly.